Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -824,19 +847,40 @@ namespace AZ::Reflection
                 // Prepare the node references for the handlers.
                 StackEntry& parentData = m_stack.back();
 
+                const AZ::Edit::ElementData* elementEditData = (classElement ? classElement->m_editData : nullptr);
                 if (classElement)
                 {
                     // Ensure our instance pointer is resolved and safe to bind to member attributes.
                     instance = AZ::Utils::ResolvePointer(instance, *classElement, *m_serializeContext);
 
                     // Since we iterate over the SerializeContext, we end up processing all elements even if they
-                    // weren't exposed to the EditContext, so we need to skip any nodes that are missing m_editData.
-                    // There are caveats:
+                    // weren't exposed to the EditContext, so we need to skip any nodes that are missing m_editData and don't have an
+                    // editDataProvider. There are caveats:
                     //   - We still need to process FLG_BASE_CLASS nodes because there could be sub-classes that are exposed to the
                     //   EditContext.
                     //   - Elements in a container will have empty m_editData, so we need to still include nodes who are members of
                     //   containers.
-                    if (!classElement->m_editData && ((classElement->m_flags & AZ::SerializeContext::ClassElement::FLG_BASE_CLASS) == 0) &&
+
+                    if (classData && !m_stack.empty())
+                    {
+                        for (auto rIter = m_stack.rbegin(), rEnd = m_stack.rend(); (rIter != rEnd && !elementEditData); ++rIter)
+                        {
+                            auto& ancestorData = *rIter;
+
+                            if (ancestorData.m_classData && ancestorData.m_classData->m_editData &&
+                                ancestorData.m_classData->m_editData->m_editDataProvider)
+                            {
+                                const AZ::Edit::ElementData* overrideData = ancestorData.m_classData->m_editData->m_editDataProvider(
+                                    ancestorData.m_instance.m_address, instance, classData->m_typeId);
+                                if (overrideData)
+                                {
+                                    elementEditData = overrideData;
+                                }
+                            }
```

Comment:
Just confirming my understanding: this is going from the top to the bottom of the stack to ultimately get the most ancestral override data, right? So this would be equivalent to forward iterating, finding the first instance of override data, and halting iteration?

Reply:
no, it's going from the bottom to the top (most specific to least specific, and it does halt iteration in the for loop when it finds valid elementData here:
(rIter != rEnd && !elementEditData);